src/lib/dlt_user.c

```diff
@@ -4972,22 +4972,21 @@ int dlt_start_threads()
 
     /* wait at most 5s */
     gettimeofday(&now,NULL);
-    time_to_wait.tv_sec = now.tv_sec+5;
+    time_to_wait.tv_sec = now.tv_sec+10;
     time_to_wait.tv_nsec = 0;
 
-     /* 
+     /*
      * wait until the house keeper is up and running
      * use the predicate to protect against spurious wake ups
      * */
-    while (!dlt_housekeeper_running) {
-        signal_status = pthread_cond_timedwait(
-            &dlt_housekeeper_running_cond, 
-            &dlt_housekeeper_running_mutex,
-            &time_to_wait);
-        if (signal_status != 0) {
-            dlt_log(LOG_CRIT, "Failed to wait for house keeper thread!\n");
-        }
+    signal_status = pthread_cond_timedwait(
+        &dlt_housekeeper_running_cond, 
+        &dlt_housekeeper_running_mutex,
+        &time_to_wait);
+    if (signal_status != 0) {
+        dlt_log(LOG_CRIT, "Failed to wait for house keeper thread!\n");
```

Comment:
If signal_status != 0, the housekeeper thread is not running correctly. 
I would propose to call dlt_stop_threads() the thread and to return a "-1" here as well.

dlt_stop_threads() should also be called in case dlt_user_trace_network_sgmented_thread could not be started.

This ensures that we leave DLT does not keep any threads running in case it returns an error code.

Reply:
good idea :+1: updated the commit

Reply:
okay so I had to make some more changes, especially we have to take into account the scenario that the clock moves in while we're in this loop. 
To prevent issues with that CLOCK_MONONTIC_RAW is used. 
Reasoning for the implementation and why I'm using the loop again is given in the code.

Reply:
oh and one more edit... I'm using `CLOCK_MONOTONIC` instead of `CLOCK_MONONTIC_RAW` because `RAW` is not supported by some variants of glibc and it will return `EINVAL` or even call `abort` in some circumstances. But only if it actually has to wait on the cond var, that's why I didn't catch it in the first version but only after some extensive testing.